File: day14/python/part2.py
import time


def calculate(path):
    with open(path, "r") as f:
        lines = f.readlines()
        result = 0
        for line in lines:
            # BEGIN TO DELETE
            result = len(line)
            # END TO DELETE
        return str(result)


if __name__ == "__main__":
    start = time.time_ns()
    result = calculate("../input.txt")
    end = time.time_ns()

    print(f"Result:\n{result.rstrip(" \t\r\n")}")
    print(f"Execution time: {(end - start)/1000000:.3f} ms")

File: day14/node/part2.js
import fs from "fs"
import { performance } from "perf_hooks"
import * as url from 'node:url';

export function calculate(path) {
  const fullFileContents = fs.readFileSync(path, 'utf8')
  const lines = fullFileContents.trim().split("\n")

  let result = 0

  for (let i = 0; i < lines.length; i++) {
    const line = lines[i]

    // BEGIN TO DELETE
    result = line.length
    // END TO DELETE
  }

  return result.toString()
}

if (process.argv[1] === url.fileURLToPath(import.meta.url)) {
  const start = performance.now()
  let result = calculate("../input.txt")
  const end = performance.now()

  console.log(`Result:\n${result.trimEnd()}`)
  console.log(`Execution time: ${(end - start).toFixed(3)} ms`)
}
